test cases/unit/59 add c17 to c_std/read_env.c

```diff
@@ -0,0 +1,20 @@
+/* Print the values of environment variables _compiler and _std. */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+int main(int argc,char **argv) {
+    if (getenv("_compiler") == NULL || getenv("_std") == NULL) {
+        printf("ERROR   : Test failed\n");
+        exit(1);
+    } else {
+#if defined(__STDC_VERSION__)
+        printf("NOTICE  : Test using %s -std=%s passed, __STDC_VERSION__=%ldL\n",
+            getenv("_compiler"), getenv("_std"), __STDC_VERSION__);
+#else
+        printf("NOTICE  : Test using %s -std=%s passed, standard=%s\n",
+            getenv("_compiler"), getenv("_std"), "ANSI-X3.159-1989");
+#endif
```

Comment:
This test isn't actually doing anything useful, it's just testing that the environment was properly passed. This is actually a really hard thing to test in this way. I think I'd be okay with just not having a test, or if you want to add a unit test that generates one target per standard and then introspecting the targets to make sure that the correct `-std=` was added.

Honestly though, I think we have pretty good coverage for the standards in use already so I'd just drop the test :)

Reply:
I'm agreeable with that.  I'll remove that test directory.

This test was primarily for me to check that my changes are working without breaking Meson. There shouldn't be a need for a standard regression test of this kind.  If the c_std mechanism stops working altogether, then I'm sure other regression tests will detect that.

You're right, I don't actually provide automated result checking to see if a compile actually used the specified standard level. I've done that manually by looking at the __STDC_VERSION__ outputs and the ninja -v compile lines.  For my sanity testing, if the test executes and returns exit status 0 then at least meson, ninja, and the target compiler have not aborted due to the standard request. I made the outputs vary per test just to match them up in my manual inspection.  That's not really the way a long-term regression test should work.

Thanks again for all the help.

Reply:
Unit test was removed in the branch. I'll marked this one resolved.